### PR TITLE
Update PluginFactory.sol

### PR DESCRIPTION
**Pull request by 0xScratch**
Same as #81

### CODE REVIEW

File: darc-protocol/contracts/protocol/Plugin/PluginFactory.sol [hunk 1]
```diff
@@ -126,27 +126,37 @@ contract PluginFactory is ConditionExpressionFactory{
...
       }
 
       // construct the result for each logical operator and return
       if (loType == EnumLogicalOperatorType.AND) {
         // if the operator is AND, the result is true only if all the nodes in the list are true
         bool bResult = true;
-        for (uint256 i = 0; i < validLength; i++) {
-          bResult = bResult && bResultList[i];
+        for (uint256 i; i < validLength;) {
+          bResult = bResult && bResultList[I];
```
Comment (lidangzzz) [inline darc-protocol/contracts/protocol/Plugin/PluginFactory.sol:145]: bResult = bResult && bResultList[I];
          
          I is not defined. use i instead.

**Review by lidangzzz — CHANGES_REQUESTED** [review]

File: darc-protocol/contracts/protocol/Plugin/PluginFactory.sol (hunk 1 above)
Comment (Aryan9592) [reply darc-protocol/contracts/protocol/Plugin/PluginFactory.sol:145]: Sure!

**Review by l-301 — APPROVED** [review]

**Comment by lidangzzz** [comment]
I'm sorry, but optimizing the loop gas is not the highest priority since the whole DARC protocol may consume 10x~100x gas than regular smart contract operation. We will have an overall roadmap about optimization in the future. I will just leave this pull request open until other optimization works are ready. Sorry for my late reply.

**Comment by Aryan9592** [comment]
Sure @lidangzzz

**Pull request closed without merging**